### PR TITLE
Fix infinite recursion in FirebaseStorageError message getter

### PR DESCRIPTION
**Pull request by mitchellwills**
This was originally broken in the refactor in #3945

### CODE REVIEW

File: packages/storage/src/implementation/error.ts [hunk 1]
```diff
@@ -27,16 +27,15 @@ export class FirebaseStorageError extends FirebaseError {
    */
   customData: { serverResponse: string | null } = { serverResponse: null };
 
+  private readonly baseMessage: string;
```
Comment (hsubox76) [inline packages/storage/src/implementation/error.ts:30]: `_baseMessage`

File: .changeset/long-forks-bow.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+'firebase': patch
+'@firebase/storage': patch
+---
+
+Fix infinite recursion in FirebaseStorageError message getter
```
Comment (hsubox76) [inline .changeset/long-forks-bow.md:6]: Is there an associated Github issue? Or an example of when the problem comes up in usage? So I can better explain to users in the release notes.

File: packages/storage/src/implementation/error.ts (hunk 1 above)
Comment (mitchellwills) [reply packages/storage/src/implementation/error.ts]: Done

File: .changeset/long-forks-bow.md (hunk 2 above)
Comment (mitchellwills) [reply .changeset/long-forks-bow.md:6]: No there isn't an existing issue (beyond this PR). Basically any time the `message` getter is accessed you would get a `Maximum call stack size exceeded` Error raised.

**Review by hsubox76 — APPROVED** [review]

**Review by hsubox76 — CHANGES_REQUESTED** [review]
Sorry, I thought the test failures were flaky and missed the actual problem. `super()` needs to be called with the message or else when the error is thrown it doesn't actually throw with the message. I don't think the property can be overridden with a getter. I think we should keep `_baseMessage`, get rid of `get message()`, and add some code to `set serverResponse` to either set `this.message` to `_baseMessage + serverResponse` or `_baseMessage` depending on the arg.

**Comment by mitchellwills** [comment]
Superseded by #4807

**Pull request closed without merging**